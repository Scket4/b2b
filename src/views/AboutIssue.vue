<template>
	<div class="wrapper">
		<router-link to="/"><p class="backPage">Назад</p></router-link>
		<div class="about-issue">
			<div class="about-issue__header">
				<div class="about-issue__header--user">
					{{ issueInfo.user.login }}
				</div>
				<p>написал</p>
				<div class="about-issue__header--date">{{ createdDate }}</div>
			</div>
			<main class="about-issue__main">
				<p class="about-issue__main--text" ref="mainText"></p>
			</main>
		</div>
	</div>
</template>

<script>
import { formatDate } from '../utils/formatDate';

const showdown = require('showdown'),
	converter = new showdown.Converter();

export default {
	name: 'AboutIssue',
	props: {
		issue: String
	},
	data() {
		return {
			issueInfo: null
		};
	},
	beforeMount() {
		this.issueInfo = JSON.parse(this.issue);
	},
	mounted() {
		this.$refs.mainText.innerHTML = this.mainText;
	},
	computed: {
		createdDate() {
			return formatDate(this.issueInfo.created_at);
		},
		mainText() {
			return converter.makeHtml(this.issueInfo.body);
		}
	}
};
</script>

<style lang="sass">
@import '../variables/variables.sass'

.wrapper
  width: 100%
  display: flex
  justify-content: center
  padding-top: 200px
  position: relative

.about-issue
  width: 50%
  border: 1px solid $grey
  border-radius: 15px
  &__header
    width: 100%
    height: 60px
    padding: 0 30px
    background-color: $low-blue-bg
    border-radius: 15px
    border-bottom: 1px solid $grey
    display: flex
    justify-content: flex-start
    align-items: center
    &--user,
    &--date
      margin: 0 10px
  &__main
    padding: 20px 30px
    overflow: hidden
    &--text
      line-height: 30px

.backPage
  position: absolute
  font-size: 26px
  cursor: pointer
  top: 20px
  left: 20px

@media (max-width: 1024px)
  .about-issue
    width: 90%
</style>