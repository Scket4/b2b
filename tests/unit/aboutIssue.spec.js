import { mount, createLocalVue } from '@vue/test-utils';
import AboutIssue from '@/views/AboutIssue';

const localVue = createLocalVue();
const issue = {
	title: 'Ошибка в асинхронных операциях vue',
	created_at: '2012-10-09T23:39:01Z',
	user: {
		login: 'alxxsck'
	},
	body: '### Some markDown text ###'
};

describe('unit tests for issues page', () => {
	it('should init correctly', () => {
		const wrapper = mount(AboutIssue, {
			localVue,
			propsData: { issue: JSON.stringify(issue) }
		});

		expect(wrapper.isVueInstance).toBeTruthy();
		expect(wrapper.is(AboutIssue)).toBeTruthy();
		expect(wrapper.vm.issue).toEqual(JSON.stringify(issue));
		expect(wrapper.vm.createdDate).toEqual('09.10.2012');
		expect(wrapper.vm.mainText).toEqual(
			'<h3 id="somemarkdowntext">Some markDown text</h3>'
		);
	});

	it('should render correctly', () => {
		const wrapper = mount(AboutIssue, {
			localVue,
			propsData: { issue: JSON.stringify(issue) }
		});

		expect(wrapper.find('.about-issue__header--user').text()).toEqual(
			issue.user.login
		);
		expect(wrapper.find('h3').text()).toEqual('Some markDown text');
	});

	it('should match the snapshot', () => {
		const wrapper = mount(AboutIssue, {
			localVue,
			propsData: { issue: JSON.stringify(issue) }
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
