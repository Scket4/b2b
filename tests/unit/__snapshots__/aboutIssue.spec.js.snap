// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for issues page should match the snapshot 1`] = `
<div class="wrapper">
  <p class="backPage">Назад</p>
  <div class="about-issue">
    <div class="about-issue__header">
      <div class="about-issue__header--user">
        alxxsck
      </div>
      <p class="isRight">написал</p>
      <div class="about-issue__header--date">09.10.2012</div>
    </div>
    <main class="about-issue__main">
      <p class="about-issue__main--text">
      <h3 id="somemarkdowntext">Some markDown text</h3>
      </p>
    </main>
  </div>
</div>
`;
